src/app/stores/reducer.ts

```diff
@@ -728,21 +729,32 @@ export function environmentReducer(
 
     case ActionTypes.ADD_ROUTE_RESPONSE: {
       const newRouteResponse: RouteResponse = action.routeReponse;
-
+      const newUUID = uuid();
+      const activeRouteResponseUUID = state.activeRouteResponseUUID === newRouteResponse.uuid ?
+        newUUID : newRouteResponse.uuid;
       newState = {
         ...state,
-        activeRouteResponseUUID: newRouteResponse.uuid,
+        activeRouteResponseUUID,
         activeTab: 'RESPONSE',
         environments: state.environments.map((environment) => {
           if (environment.uuid === state.activeEnvironmentUUID) {
             return {
               ...environment,
               routes: environment.routes.map((route) => {
                 if (route.uuid === state.activeRouteUUID) {
-                  return {
-                    ...route,
-                    responses: [...route.responses, newRouteResponse]
-                  };
+                  let responses = [...route.responses];
+                  if (state.activeRouteResponseUUID === newRouteResponse.uuid) {
+                    const activeRouteResponseIndex = route.responses.findIndex(
+                      (routeResponse: RouteResponse) => {
+                        return routeResponse.uuid === state.activeRouteResponseUUID;
+                      }
+                    );
+                    newRouteResponse.uuid = newUUID;
+                    responses.splice(activeRouteResponseIndex + 1, 0, newRouteResponse);
+                  } else {
+                    responses.push(newRouteResponse);
+                  }
+                  return { ...route, responses };
```

Comment:
Please implements the same logic as the route or env duplication by providing an `afterUUID` param directly to the action (https://github.com/mockoon/mockoon/blob/master/src/app/stores/actions.ts#L235). Then you can follow the ADD_ENV implementation https://github.com/mockoon/mockoon/blob/master/src/app/stores/reducer.ts#L352